src/v/kafka/server/handlers/fetch.cc

```diff
@@ -226,12 +226,12 @@ static read_result::memory_units_t reserve_memory_units(
   ssx::semaphore& memory_sem,
   ssx::semaphore& memory_fetch_sem,
   const size_t max_bytes,
+  const size_t max_batch_size,
```

Comment:
missing doc in function doc

Reply:
Added the function doc for it. And also simplified the logic of the function to make it easier to understand in isolation.